crates/pipeline_manager/src/pipeline_automata.rs

```diff
@@ -567,52 +562,34 @@ impl<T: PipelineExecutor> PipelineAutomaton<T> {
     /// places pipeline in the `Failed` state to avoid instant automatic
     /// restart.  Otherwise (the desired state was `Shutdown`), set the
     /// state of the pipeline to `Shutdown`.
-    async fn force_kill_pipeline<E>(
+    async fn mark_pipeline_as_failed<E>(
```

Comment:
There should still be a way to actually kill the pipeline process, at least for the local runner. Otherwise, how do we get rid or unresponsive pipelines?

Reply:
@ryzhyk Calling /shutdown via the public API would do exactly that. It'll change the desired state to shutdown, which will immediately call runner.shutdown(), which for the local runner will set the pipeline_handle to None like before (which should kill the process).